docs/source/app-dev/app-arch.rst

```diff
@@ -1,295 +1,120 @@
+| `@daml/react <https://www.npmjs.com/package/@daml/react>`_   | React hooks to query/create/exercise DAML contracts                      |
++--------------------------------------------------------------+--------------------------------------------------------------------------+
+| `@daml/ledger <https://www.npmjs.com/package/@daml/ledger>`_ | DAML ledger object to connect and directly submit commands to the ledger |
++--------------------------------------------------------------+--------------------------------------------------------------------------+
+| `@daml/types  <https://www.npmjs.com/package/@daml/types>`_  | DAML data types                                                          |
++--------------------------------------------------------------+--------------------------------------------------------------------------+
 
-Examples of these are automation and interactive applications that let a user or code react to complex state on the ledger (e.g. the DA Navigator tool).
+You can install any of these libraries by running ``yarn add @daml/react`` in the ``ui`` directory
+of your project. Please explore the ``create-daml-app`` example project to see the usage of these
+libraries.
 
-Which approach to take
-======================
+The interface between the frontend and the backend of your application is defined by your DAML code.
+To make your life easy when interacting with the ledger, the DAML assistant can generate
+corresponding typescript data definitions for the data types declared in DAML.
 
-For all except the simplest applications, we generally recommend the state-driven approach. State-driven applications are easier to reason about when determining correctness, so this makes design and implementation easier.
+.. code-block:: bash
 
-In practice, most applications are actually a mixture of the two styles, with one predominating. It is easier to add some event handling to a state-driven application, so it is better to start with that style.
+  daml codegen ts .daml/dist/<your-project-name.dar> -o daml2ts
 
-Structuring an application
-**************************
+This will create a typescript project under ``daml2ts`` that you can install by changing directory
+to ``daml2ts`` and running ``yarn install``. We recommend using the provided ``daml-start.sh``
+script at the root of the project directory of ``create-daml-app``. This will take care of creating
+and updating the ``daml2ts`` library whenever you change your DAML model automatically.
 
-Although applications that communicate with the ledger have many purposes, they generally have some common features, usually related to their style: event-driven or state-driven. This section describes these commonalities, and the major functions of each of these styles.
+If you choose a different frontend framework, the DAML assistant will also help you generate code
+for a variety of other programming languages. Run ``daml codegen --help`` to find out more.
```

Comment:
If I run "daml codegen --help" it tells me that I can run it for scala, java, and ts. So I can't use it to generate	vue.js code or express.js code (frontend frameworks). Would remove this part and would describe that daml2ts provides data types for other frontend frameworks/libraries that could be used instead of React, e.g., Angular.

Reply:
changed it, not mentioning the other codegen's anymore.